src/change_stream.ts

```diff
@@ -240,18 +250,30 @@ export class ChangeStream extends EventEmitter {
     // Listen for any `change` listeners being added to ChangeStream
     this.on('newListener', (eventName: string) => {
       if (eventName === 'change' && this.cursor && this.listenerCount('change') === 0) {
-        this.cursor.on('data', change => processNewChange(this, change));
+        this.streamEvents(this.cursor);
       }
     });
 
-    // Listen for all `change` listeners being removed from ChangeStream
     this.on('removeListener', (eventName: string) => {
       if (eventName === 'change' && this.listenerCount('change') === 0 && this.cursor) {
-        this.cursor.removeAllListeners('data');
+        this[kCursorStream]?.removeAllListeners('data');
       }
     });
   }
 
+  /** @internal */
+  streamEvents(cursor: ChangeStreamCursor): void {
```

Comment:
it doesn't look like this method is required outside of this implementation, is there a reason you attached it to the `ChangeStream` class? Prefer to make it a file-local private method than attaching it as public API (even if its marked internal).